src/Dependencies/Collections/ImmutableSegmentedDictionary`2+Builder.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    internal readonly partial struct ImmutableSegmentedDictionary<TKey, TValue>
+    {
+        public sealed partial class Builder : IDictionary<TKey, TValue>, IReadOnlyDictionary<TKey, TValue>, IDictionary
+        {
+            private readonly ImmutableSegmentedDictionary<TKey, TValue> _dictionary;
```

Comment:
Consider naming this something like `_baseDictionary` or the like, to indicate it's the base dictionary this builder is based on.

Reply:
This is now commented